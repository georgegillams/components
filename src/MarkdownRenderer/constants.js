const DEFAULT_SUPPORTED_FEATURES = [
  'code',
  'blockCode',
  'bold',
  'italic',
  'strikethrough',
  'quotation',
  'bigLink',
  'link',
  'footnote',
  'image',
  'video',
  'citation',
  'sectioning',
  'linebreak',
];

export default DEFAULT_SUPPORTED_FEATURES;
export { DEFAULT_SUPPORTED_FEATURES };
