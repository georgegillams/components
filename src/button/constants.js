export const BUTTON_TYPES = {
  primary: 'primary',
  destructive: 'destructive',
  bouncy: 'bouncy',
};

export const BUTTON_SIZES = {
  regular: 'regular',
  large: 'large',
};
