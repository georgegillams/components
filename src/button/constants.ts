import { capitalise } from '../helpers/helper-functions';

export enum BUTTON_TYPES {
  primary = 'primary',
  secondary = 'secondary',
  bouncy = 'bouncy',
  destructive = 'destructive',
}

export enum BUTTON_SIZES {
  regular = 'regular',
  large = 'large',
}

export const buttonThemeProperties = Object.keys(BUTTON_TYPES).reduce(
  (prev, next) => [
    ...prev,
    `button${capitalise(next)}BackgroundColor`,
    `button${capitalise(next)}BackgroundColorDarkMode`,
    `button${capitalise(next)}ForegroundColor`,
    `button${capitalise(next)}ForegroundColorDarkMode`,
    `button${capitalise(next)}ForegroundColorHover`,
    `button${capitalise(next)}ForegroundColorHoverDarkMode`,
  ],
  [] as string[],
);
