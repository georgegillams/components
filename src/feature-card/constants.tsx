export enum FEATURE_CARD_LAYOUTS {
  auto = 'auto',
  narrowCompact = 'narrowCompact',
}
