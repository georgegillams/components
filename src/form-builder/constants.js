export const FORM_FIELD_VISIBILITY = {
  VISIBLE: 'visible',
  VISUALLY_HIDDEN: 'visually_hidden',
  OFF: 'off',
};
