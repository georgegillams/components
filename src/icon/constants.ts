export interface IconProps extends React.HTMLAttributes<HTMLOrSVGElement> {
  large?: boolean;
}
