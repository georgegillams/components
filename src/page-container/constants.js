export const WIDTHS = {
  default: 'default',
  prose: 'prose',
  fullWidth: 'fullWidth',
};
