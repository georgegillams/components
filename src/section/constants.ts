import PropTypes from 'prop-types';

export const commonPropTypes = {
  anchor: PropTypes.bool,
  padding: PropTypes.bool,
  name: PropTypes.string,
  children: PropTypes.node,
  textClassName: PropTypes.string, // For backwards compatibility
  headingProps: PropTypes.object, // For backwards compatibility
};

export const commonDefaultProps = {
  anchor: false,
  name: null,
  padding: true,
  children: null,
  textClassName: null,
  headingProps: null,
};
