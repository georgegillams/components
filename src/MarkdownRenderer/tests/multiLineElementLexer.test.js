#!/usr/bin/env node

import {
  parseMultiLineElements,
  parseLinesForQuoteBlock,
  parseLinesForBulletList,
} from '../multiLineElementLexer.js';
import { DEFAULT_SUPPORTED_FEATURES } from '../constants.js';

// #region individual lists
test('parses lines to insert bullet-list', () => {
  const result = parseLinesForBulletList([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: ' - Bullet 1' },
    { type: 'line', content: ' - Bullet 2' },
    { type: 'line', content: ' - Bullet 3' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert numbered-list', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: ' 1. Numbered item 1' },
    { type: 'line', content: ' 2. Numbered item 2' },
    { type: 'line', content: ' 3. Numbered item 3' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines containing no comment block', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
    { type: 'line', content: 'Line 4' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert comment block at start', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: '> Quote line 1' },
    { type: 'line', content: '> Quote line 2' },
    { type: 'line', content: '> Quote line 3' },
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert comment block in middle', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: '> Quote line 1' },
    { type: 'line', content: '> Quote line 2' },
    { type: 'line', content: '> Quote line 3' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
// #endregion individual lists

// #region structure
test('parses comment block inside section', () => {
  const result = parseMultiLineElements(
    [
      { type: 'line', content: 'Line 1' },
      {
        type: 'section',
        children: [
          { type: 'line', content: 'Line 2' },
          { type: 'line', content: '> Quote line 1' },
          { type: 'line', content: '> Quote line 2' },
          { type: 'line', content: '> Quote line 3' },
          { type: 'line', content: 'Line 3' },
        ],
      },
      { type: 'line', content: 'Line 4' },
    ],
    DEFAULT_SUPPORTED_FEATURES,
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('ignores comment block if feature is unsupported', () => {
  const result = parseMultiLineElements(
    [
      { type: 'line', content: 'Line 1' },
      { type: 'line', content: 'Line 2' },
      { type: 'line', content: '> Unsupported quote line 1' },
      { type: 'line', content: '> Unsupported quote line 2' },
      { type: 'line', content: '> Unsupported quote line 3' },
      { type: 'line', content: 'Line 3' },
    ],
    [],
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
// #endregion structure
