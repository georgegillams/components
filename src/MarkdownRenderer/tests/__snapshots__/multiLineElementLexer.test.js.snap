// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores comment block if feature is unsupported 1`] = `
Array [
  Object {
    "children": Array [
      "Line 1",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "Line 2",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "> Unsupported quote line 1",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "> Unsupported quote line 2",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "> Unsupported quote line 3",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "Line 3",
    ],
    "type": "paragraph",
  },
]
`;

exports[`parses comment block inside section 1`] = `
Array [
  Object {
    "children": Array [
      "Line 1",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Line 2",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Quote line 1",
          "Quote line 2",
          "Quote line 3",
        ],
        "type": "quotation",
      },
      Object {
        "children": Array [
          "Line 3",
        ],
        "type": "paragraph",
      },
    ],
    "type": "section",
  },
  Object {
    "children": Array [
      "Line 4",
    ],
    "type": "paragraph",
  },
]
`;

exports[`parses lines containing no comment block 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
  Object {
    "content": "Line 4",
    "type": "line",
  },
]
`;

exports[`parses lines to insert comment block at start 1`] = `
Array [
  Object {
    "children": Array [
      "Quote line 1",
      "Quote line 2",
      "Quote line 3",
    ],
    "type": "quotation",
  },
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert comment block in middle 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "children": Array [
      "Quote line 1",
      "Quote line 2",
      "Quote line 3",
    ],
    "type": "quotation",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;
